go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go

```diff
@@ -77,19 +78,29 @@ type TxThrottler struct {
 	state *txThrottlerState
 
 	target *querypb.Target
+
+	// stats
+	throttlerRunning  *stats.Gauge
+	requestsTotal     *stats.Counter
+	requestsThrottled *stats.Counter
 }
 
+// TxThrottlerName is the name the wrapped go/vt/throttler object will be registered with
+// go/vt/throttler.GlobalManager.
+const TxThrottlerName = "TransactionThrottler"
+
 // NewTxThrottler tries to construct a TxThrottler from the
 // relevant fields in the tabletenv.Config object. It returns a disabled TxThrottler if
 // any error occurs.
 // This function calls tryCreateTxThrottler that does the actual creation work
 // and returns an error if one occurred.
 func NewTxThrottler(config *tabletenv.TabletConfig, topoServer *topo.Server) *TxThrottler {
-	txThrottler, err := tryCreateTxThrottler(config, topoServer)
+	env := tabletenv.NewEnv(config, TxThrottlerName)
+	txThrottler, err := tryCreateTxThrottler(env, topoServer)
```

Comment:
instead of receiving `config *tabletenv.TabletConfig` can be changed to `env tabletenv.Env` and `tabletserver` can be passed in as it implemented the interface.

`tsv.txThrottler = txthrottler.NewTxThrottler(tsv.config, topoServer)` -> `tsv.txThrottler = txthrottler.NewTxThrottler(tsv, topoServer)`

Reply:
@harshit-gangal unfortunately that creates an import cycle because `tabletserver` imports `txthrottler`, and this change would cause `txthrottler` to import `tabletserver`

Reply:
tx_throttler will have `tabletenv` as the import and not the `tabletserver`

Reply:
I see that inside `tabletenv` package `config.go` imports `throttler` which seems very wrong to me. Probably a code that needs some refactoring.

Just trying to make things more clean and similar.